### PR TITLE
docs/get-started-with-wp-env: diagram and links added

### PR DESCRIPTION
**Pull request by juanmaguitar**
## What?
Added a diagram to provide a quick reference to use wp-env

## Why?
To improve the learning journey of WordPress developers

### CODE REVIEW

File: docs/getting-started/devenv/get-started-with-wp-env.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ The [@wordpress/env](https://www.npmjs.com/package/@wordpress/env) package (`wp-
 
 Before following this guide, install [Node.js development tools](/docs/getting-started/devenv#node-js-development-tools) if you have not already done so.
 
+[![wp-env basics diagram](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/assets/wp-env-diagram.png)](https://excalidraw.com/#json=8Tp55B-R6Z6-pNGtmenU6,_DeBR1IBxuHNIKPTVEaseA)
```
Comment (ndiego) [inline docs/getting-started/devenv/get-started-with-wp-env.md]: I am wondering if the graphic would make more sense at the end of the Quick start section 🤔 That way the image reinforces the steps above it.

File: docs/assets/wp-env-diagram.png
Comment (ndiego) [inline docs/assets/wp-env-diagram.png:1]: We can move this to the media library on [developer.wordpress.org](http://developer.wordpress.org/) and then reference directly.

**Review by ndiego — CHANGES_REQUESTED** [review]

File: docs/getting-started/devenv/get-started-with-wp-env.md (hunk 1 above)
Comment (juanmaguitar) [reply docs/getting-started/devenv/get-started-with-wp-env.md]: I think it's more useful (and accessible) at the beginning as some sort of TL;DR

File: docs/assets/wp-env-diagram.png
Comment (juanmaguitar) [reply docs/assets/wp-env-diagram.png:1]: Done ✅

**Review by ndiego — APPROVED** [review]

File: docs/getting-started/devenv/get-started-with-wp-env.md [hunk 2]
```diff
@@ -4,6 +4,8 @@ The [@wordpress/env](https://www.npmjs.com/package/@wordpress/env) package (`wp-
 
 Before following this guide, install [Node.js development tools](/docs/getting-started/devenv#node-js-development-tools) if you have not already done so.
 
+[![wp-env basics diagram](https://developer.wordpress.org/files/2023/10/wp-env-diagram.png)](https://excalidraw.com/#json=8Tp55B-R6Z6-pNGtmenU6,_DeBR1IBxuHNIKPTVEaseA)
```
Comment (ajitbohra) [inline docs/getting-started/devenv/get-started-with-wp-env.md]: just to confirm the image will link to `https://excalidraw.com`?

Comment (juanmaguitar) [reply docs/getting-started/devenv/get-started-with-wp-env.md]: Yes, the image will link to the original diagram used to generate the image

Comment (juanmaguitar) [reply docs/getting-started/devenv/get-started-with-wp-env.md]: I have separated the link to the original diagram from the image (now it's provided in the "Additional resources" section)

**Review by ajitbohra — APPROVED** [review]

**Pull request merged**